common/client/types.go

```diff
@@ -10,6 +10,40 @@ import (
 	"github.com/smartcontractkit/chainlink/v2/common/types"
 )
 
+// RPCClient includes all the necessary generalized RPC methods along with any additional chain-specific methods.
+type RPCClient[
```

Comment:
`SendOnlyNode` and `Node` should use their own clients instead of a generalized `RPC any `type. I would rename this to `NodeClient` and use it in `multi_node.go`,  `node.go`. This is a duplicate of the following interfaces, which are now obsolete.

Reply:
And perhaps after that this can be moved under `node.go` but it might have an import implication.

Reply:
Removed all the old interfaces. We only want to use this single RPCClient interface with all of these components. `SendOnlyNode` and `Node` both use this interface for their RPC types and not `any`, which allows us to use them across chains. I don't think renaming is necessary in this case, but I could reconsider this if needed.